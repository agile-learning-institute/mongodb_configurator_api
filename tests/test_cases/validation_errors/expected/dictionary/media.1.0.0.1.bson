bsonType: object
required: ["_id", "title", "status", "last_saved"]
properties:
  _id:
    bsonType: objectId
    pattern: "^[0-9a-fA-F]{24}$"
  title:
    bsonType: string
    pattern: "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
  type:
    bsonType: string
    enum: ["video", "audio", "image", "document"]
  status:
    bsonType: string
    enum: ["draft", "active", "archived"]
  last_saved:
    bsonType: object
    required: ["from_ip", "by_user", "at_time", "correlation_id"]
    properties:
      from_ip:
        bsonType: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
      by_user:
        bsonType: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
      at_time:
        bsonType: date
      correlation_id:
        bsonType: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
  tags:
    bsonType: array
    items:
      bsonType: string
      enum: ["educational", "entertainment", "tutorial", "documentary"]
  metadata:
    bsonType: object
    properties:
      duration:
        bsonType: int
        minimum: 1
        multipleOf: 1
      format:
        bsonType: string
        enum: ["mp4", "mp3", "jpg", "pdf"]
      quality:
        bsonType: string
        enum: ["low", "medium", "high", "ultra"] 