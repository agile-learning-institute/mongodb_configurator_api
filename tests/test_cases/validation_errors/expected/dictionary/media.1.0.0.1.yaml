type: object
required: ["_id", "title", "status", "last_saved"]
properties:
  _id:
    type: string
    pattern: "^[0-9a-fA-F]{24}$"
  title:
    type: string
    pattern: "^[a-zA-Z0-9\\s.,!?-]{4,255}$"
  type:
    type: string
    enum: ["video", "audio", "image", "document"]
  status:
    type: string
    enum: ["draft", "active", "archived"]
  last_saved:
    type: object
    required: ["from_ip", "by_user", "at_time", "correlation_id"]
    properties:
      from_ip:
        type: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
      by_user:
        type: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
      at_time:
        type: string
        format: date-time
      correlation_id:
        type: string
        pattern: "^[a-zA-Z0-9_-]{4,40}$"
  tags:
    type: array
    items:
      type: string
      enum: ["educational", "entertainment", "tutorial", "documentary"]
  metadata:
    type: object
    properties:
      duration:
        type: number
        minimum: 1
        multipleOf: 1
      format:
        type: string
        enum: ["mp4", "mp3", "jpg", "pdf"]
      quality:
        type: string
        enum: ["low", "medium", "high", "ultra"] 